osu.Game.Rulesets.Rush/Objects/Drawables/DrawableRushHitObject.cs

```diff
@@ -152,10 +143,64 @@ protected override void UpdateStateTransforms(ArmedState state)
             }
         }
 
+        public virtual Drawable CreateHitExplosion() => new DefaultHitExplosion(Color4.Yellow.Darken(0.5f))
+        {
+            Anchor = Anchor,
+            Origin = Anchor.Centre,
+            Size = new Vector2(200, 200),
+            Scale = new Vector2(0.9f + RNG.NextSingle() * 0.2f),
+            Rotation = RNG.NextSingle() * 360f,
+        };
+
+        protected override JudgementResult CreateResult(Judgement judgement) => new RushJudgementResult(HitObject, (RushJudgement)judgement);
+
+        protected new void ApplyResult(Action<JudgementResult> application)
+        {
+            // This is the only point to correctly apply values to the judgement
+            // result in correct time, check whether the player collided now.
+            Action<JudgementResult> rushApplication = br =>
+            {
+                var r = (RushJudgementResult)br;
+
+                application?.Invoke(r);
+                r.PlayerCollided = drawableRuleset.PlayerCollidesWith(r.HitObject);

```

Comment:
when rewinding, results would only be reverted as seen in `DrawableHitObject.Update()` (i.e. `CheckForUpdate` is never called, thus `ApplyResult` is never called), I've updated the boolean here as it is more correct since it'll remove the *unintended* direct connection between `HealthProcessor` and `DrawableRuleset`.